### PR TITLE
Rename all singular CompilationUnit references to singular

### PR DESCRIPTION
**Pull request by louistsaitszho**
From #1947: rename almost all references of `CompilationUnit` to singular.

### CODE REVIEW

File: apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultApolloExtension.kt [hunk 1]
```diff
@@ -15,13 +15,13 @@ open class DefaultApolloExtension(val project: Project)
   val services = mutableListOf<DefaultService>()
 
   /**
-   * compilationUnits is meant to be consumed by other gradle plugin.
+   * compilationUnit is meant to be consumed by other gradle plugin.
    * The apollo plugin will add the {@link CompilationUnit} as it creates them
    */
-  internal val compilationUnits = project.container(CompilationUnit::class.java)
+  internal val compilationUnit = project.container(CompilationUnit::class.java)
```
Comment (tasomaniac) [inline apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultApolloExtension.kt]: This one is a container (which extends from Set) and internal. I think we should keep this one plural.

Comment (martinbonnin) [reply apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultApolloExtension.kt]: +1

File: apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultApolloExtension.kt [hunk 2]
```diff
@@ -15,13 +15,13 @@ open class DefaultApolloExtension(val project: Project)
...
 
-  override fun onCompilationUnits(action: Action<CompilationUnit>) {
-    compilationUnits.all(action)
```
Comment (martinbonnin) [inline apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultApolloExtension.kt]: What about keeping the old one and doing:

```
 fun onCompilationUnits(action: Action<CompilationUnit>) {
    throw Exception("please use onCompilationUnit {} instead")  
}
```

And then we remove that old version in a couple of months ?  This API is new from 1.3.0 so presumably not that many people are using it but that'll make a nicer deprecation path for the users that use it.

File: apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/api/ApolloExtension.kt [hunk 3]
```diff
@@ -39,6 +39,14 @@ interface ApolloExtension: CompilerParams {
    */
   fun onCompilationUnit(action: Action<CompilationUnit>)
 
+  /**
+   * @Deprecated
+   *
+   * Please see [onCompilationUnit]
+   */
+  @Deprecated("There is a typo. Please use onCompilationUnit instead", replaceWith = ReplaceWith("onCompilationUnit"))
```
Comment (martinbonnin) [inline apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/api/ApolloExtension.kt]: Nice :+1:

**Pull request merged**